### PR TITLE
Rewrite channel and add getters.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/monitor.toit [hunk 1]
```diff
@@ -222,12 +221,23 @@ monitor Channel:
...
 
+  /**
+  The capacity of the channel.
+  */
+  capacity -> int: return buffer_.size
+
+  /**
+  The amount of messages that are currently queued in the channel.
+  */
+  buffered_count -> int: return buffered_count_
```
Comment (kasperl) [inline lib/monitor.toit:239]: I kinda like just calling this size. What do you think?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/monitor.toit (hunk 1 above)
Comment (floitsch) [reply lib/monitor.toit:239]: I'm hesitant. I can easily see myself asking for "a channel of size 8".
To me the "size" often refers to the capacity of the channel, and not how many items are in it.

Comment (floitsch) [reply lib/monitor.toit]: Changed to `size`.
done.

**Pull request merged**